## 43% of end-end tests have passed
```
Total executed tests: 7
Failed tests: 4
Expected to fail tests: 0
Passed tests: 3
```
```
IDs of failed tests: 727231,727230,703133,727229 
```
<h3>Failed tests (4)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=ilmotta/migrated-users-must-accept-terms&PR_ID=21124&APK_NAME=21124.apk&TR_CASE_IDS=727231,727230,703133,727229">Rerun failed tests</a></li><br/><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3644453">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//android.view.ViewGroup[@content-desc='container']/android.widget.TextView[@text='Ether']/../android.widget.TextView[3]` </div><div>Device 2: `Text` is `0.03839 ETH` </div></blockquote></p>

```
critical/test_wallet.py:190: in test_wallet_send_asset_from_drawer
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Sender balance is not updated on Etherscan, it is 0.4538 but expected to be 0.454
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3644452">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//android.view.ViewGroup[@content-desc='container']/android.widget.TextView[@text='Ether']/../android.widget.TextView[3]` </div><div>Device 2: `Text` is `0.03829 ETH` </div></blockquote></p>

```
critical/test_wallet.py:159: in test_wallet_send_eth
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Sender balance is not updated on Etherscan, it is 0.4541 but expected to be 0.4543
```

<br/><br/></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3644454">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div>Device 1: Click until `EditBox` by `accessibility id`: `profile-title-input` will be presented </div></blockquote></p>

```
Test setup failed: critical/test_wallet.py:202: in prepare_devices
    self.sign_in_view.create_user()
../views/sign_in_view.py:249: in create_user
    self.generate_keys_button.click_until_presence_of_element(self.profile_title_input)
../views/base_element.py:107: in click_until_presence_of_element
    while not desired_element.is_element_displayed(1) and counter <= attempts:
../views/base_element.py:221: in is_element_displayed
    return self.wait_for_visibility_of_element(sec, ignored_exceptions=ignored_exceptions)
../views/base_element.py:145: in wait_for_visibility_of_element
    .until(expected_conditions.visibility_of_element_located((self.by, self.locator)))
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/wait.py:86: in until
    value = method(self._driver)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/expected_conditions.py:152: in _predicate
    return _element_if_visible(driver.find_element(*locator))
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webdriver.py:409: in find_element
    return self.execute(RemoteCommand.FIND_ELEMENT, {'using': by, 'value': value})['value']
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 An unknown server-side error occurred while processing the command. Original error: 'POST /element' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E   Stacktrace:
E   UnknownError: An unknown server-side error occurred while processing the command. Original error: 'POST /element' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E       at UIA2Proxy.command (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/jsonwp-proxy/proxy.js:355:13)
E       at processTicksAndRejections (node:internal/process/task_queues:95:5)
E       at AndroidUiautomator2Driver.helpers.doFindElementOrEls (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/lib/commands/find.js:30:10)
E       at doFind (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/node_modules/appium-android-driver/lib/commands/find.js:36:17)
E       at wrappedCondFn (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/timeout.ts:137:14)
E       at spin (/mnt/sauce/appium/appium-v2.0.0/node_modules/asyncbox/lib/asyncbox.js:221:20)
E       at waitForCondition (/mnt/sauce/appium/appium-v2.0.0/node_modules/asyncbox/lib/asyncbox.js:238:10)
E       at AndroidUiautomator2Driver.implicitWaitForCondition (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/timeout.ts:139:12)
E       at AndroidUiautomator2Driver.findElOrEls (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/node_modules/appium-android-driver/lib/commands/find.js:60:5)
E       at AndroidUiautomator2Driver.findElOrElsWithProcessing (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/find.ts:60:12)
E       at AndroidUiautomator2Driver.findElement (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/find.ts:75:12)
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/808b808577e84b39ac9fe7d09ebf6d3e?auth=5cf3e50c90ad0e4d2ed8fb032bdeca29">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3644451">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div># STEP: Check that removed user is not shown in the list anymore</div><div>Device 1: Wait for element `Button` for max 30s and click when it is available </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:239: in test_restore_multiaccount_with_waku_backup_remove_switch
    self.sign_in.show_profiles_button.wait_and_click()
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: Button by accessibility id:`show-profiles` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7d879cc70d9f43d78eea21ad13cb0f3a?auth=61b5eff7fc718f0fec84ec8ad9ccbc5c#82">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (3)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3644449">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/35b8e4f5fcd9477dae504067ff451d55?auth=24fc9e5601e3f30f241903c2e1d83a9f#101">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d224187518954e198ee474d26e314216?auth=d2ad0ec152f4bb10088e9a9ffe53914f#90">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3644450">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8e223316d5a34d5d83f8f44e828f1165?auth=82884a865ceffc5e20ce581afb71825f#168">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/308071d0228249bb8aa22bfaa3fa3a00?auth=8fcfce84a11c4792efe396faeb98e4d7#116">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3644448">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7d879cc70d9f43d78eea21ad13cb0f3a?auth=61b5eff7fc718f0fec84ec8ad9ccbc5c#53">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/modules/react-native-status/android/src/main/java/im/status/ethereum/module/AccountManager.kt
+++ b/modules/react-native-status/android/src/main/java/im/status/ethereum/module/AccountManager.kt
@@ -253,6 +253,11 @@ class AccountManager(private val reactContext: ReactApplicationContext) : ReactC
         utils.executeRunnableStatusGoMethod({ Statusgo.initializeApplication(request) }, callback)
     }
 
+    @ReactMethod
+    private fun acceptTerms(callback: Callback) {
+        Log.d(TAG, "acceptTerms")
+        utils.executeRunnableStatusGoMethod({ Statusgo.acceptTerms() }, callback)
+    }
 
     @ReactMethod
     fun logout() {

--- a/modules/react-native-status/ios/RCTStatus/AccountManager.m
+++ b/modules/react-native-status/ios/RCTStatus/AccountManager.m
@@ -204,6 +204,14 @@ RCT_EXPORT_METHOD(initializeApplication:(NSString *)request
     callback(@[result]);
 }
 
+RCT_EXPORT_METHOD(acceptTerms:(RCTResponseSenderBlock)callback) {
+#if DEBUG
+    NSLog(@"acceptTerms() method called");
+#endif
+    NSString *result = StatusgoAcceptTerms();
+    callback(@[result]);
+}
+
 RCT_EXPORT_METHOD(openAccounts:(RCTResponseSenderBlock)callback) {
 #if DEBUG
     NSLog(@"OpenAccounts() method called");

--- a/modules/react-native-status/nodejs/status.cpp
+++ b/modules/react-native-status/nodejs/status.cpp
@@ -761,6 +761,26 @@ void _Logout(const FunctionCallbackInfo<Value>& args) {
 
 }
 
+void _AcceptTerms(const FunctionCallbackInfo<Value>& args) {
+	Isolate* isolate = args.GetIsolate();
+
+	if (args.Length() != 0) {
+		// Throw an Error that is passed back to JavaScript
+		isolate->ThrowException(Exception::TypeError(
+			String::NewFromUtf8Literal(isolate, "Wrong number of arguments for AcceptTerms")));
+		return;
+	}
+
+	// Check the argument types
+
+	// Call exported Go function, which returns a C string
+	char *c = AcceptTerms();
+
+	Local<String> ret = String::NewFromUtf8(isolate, c).ToLocalChecked();
+	args.GetReturnValue().Set(ret);
+	delete c;
+}
+
 void _HashMessage(const FunctionCallbackInfo<Value>& args) {
 	Isolate* isolate = args.GetIsolate();
         Local<Context> context = isolate->GetCurrentContext();
@@ -1998,6 +2018,7 @@ void init(Local<Object> exports) {
 	NODE_SET_METHOD(exports, "multiAccountStoreAccount", _MultiAccountStoreAccount);
 	NODE_SET_METHOD(exports, "initKeystore", _InitKeystore);
 	NODE_SET_METHOD(exports, "initializeApplication", _InitializeApplication);
+	NODE_SET_METHOD(exports, "acceptTerms", _AcceptTerms);
 	NODE_SET_METHOD(exports, "fleets", _Fleets);
 	NODE_SET_METHOD(exports, "stopCPUProfiling", _StopCPUProfiling);
 	NODE_SET_METHOD(exports, "encodeTransfer", _EncodeTransfer);

--- a/src/native_module/core.cljs
+++ b/src/native_module/core.cljs
@@ -79,6 +79,12 @@
                           (types/clj->json request)
                           #(callback (types/json->clj %))))
 
+(defn accept-terms
+  ([]
+   (native-utils/promisify-native-module-call accept-terms))
+  ([callback]
+   (.acceptTerms ^js (account-manager) callback)))
+
 (defn prepare-dir-and-update-config
   [key-uid config callback]
   (log/debug "[native-module] prepare-dir-and-update-config")

--- a/src/status_im/contexts/onboarding/intro/view.cljs
+++ b/src/status_im/contexts/onboarding/intro/view.cljs
@@ -11,66 +11,92 @@
     [utils.i18n :as i18n]
     [utils.re-frame :as rf]))
 
+(defn- show-terms-of-use
+  []
+  (rf/dispatch [:show-bottom-sheet {:content terms/terms-of-use :shell? true}]))
+
+(defn- show-privacy-policy
+  []
+  (rf/dispatch [:show-bottom-sheet {:content privacy/privacy-statement :shell? true}]))
+
+(defn- terms
+  [terms-accepted? set-terms-accepted?]
+  [rn/view {:style style/terms-privacy-container}
+   [rn/view
+    {:accessibility-label :terms-privacy-checkbox-container}
+    [quo/selectors
+     {:type      :checkbox
+      :blur?     true
+      :checked?  terms-accepted?
+      :on-change #(set-terms-accepted? not)}]]
+   [rn/view {:style style/text-container}
+    [quo/text
+     {:style style/plain-text
+      :size  :paragraph-2}
+     (str (i18n/label :t/accept-status-tos-prefix) " ")]
+    [quo/text
+     {:on-press show-terms-of-use
+      :style    style/highlighted-text
+      :size     :paragraph-2
+      :weight   :medium}
+     (i18n/label :t/terms-of-service)]
+    [quo/text
+     {:style style/plain-text
+      :size  :paragraph-2}
+     " " (i18n/label :t/and) " "]
+    [quo/text
+     {:on-press show-privacy-policy
+      :style    style/highlighted-text
+      :size     :paragraph-2
+      :weight   :medium}
+     (i18n/label :t/intro-privacy-policy)]]])
+
+(defn- explore-new-status
+  []
+  (rf/dispatch [:profile/explore-new-status]))
+
+(defn- sync-or-recover-profile
+  []
+  (when-let [blur-show-fn @overlay/blur-show-fn-atom]
+    (blur-show-fn))
+  (rf/dispatch [:open-modal :screen/onboarding.sync-or-recover-profile]))
+
+(defn- create-profile
+  []
+  (when-let [blur-show-fn @overlay/blur-show-fn-atom]
+    (blur-show-fn))
+  (rf/dispatch [:open-modal :screen/onboarding.new-to-status]))
+
 (defn view
   []
-  (let [[terms-accepted? set-terms-accepted?] (rn/use-state false)]
+  (let [[terms-accepted? set-terms-accepted?] (rn/use-state false)
+        from-v1-without-terms-accepted?       (rf/sub [:profile/from-status-v1-without-terms-accepted?])]
     [rn/view {:style style/page-container}
      [background/view false]
      [quo/bottom-actions
-      {:container-style      (style/bottom-actions-container (safe-area/get-bottom))
-       :actions              :two-vertical-actions
-       :description          :top
-       :description-top-text [rn/view
-                              {:style style/terms-privacy-container}
-                              [rn/view
-                               {:accessibility-label :terms-privacy-checkbox-container}
-                               [quo/selectors
-                                {:type      :checkbox
-                                 :blur?     true
-                                 :checked?  terms-accepted?
-                                 :on-change #(set-terms-accepted? not)}]]
-                              [rn/view {:style style/text-container}
-                               [quo/text
-                                {:style style/plain-text
-                                 :size  :paragraph-2}
-                                (str (i18n/label :t/accept-status-tos-prefix) " ")]
-                               [quo/text
-                                {:on-press #(rf/dispatch [:show-bottom-sheet
-                                                          {:content terms/terms-of-use
-                                                           :shell?  true}])
-                                 :style    style/highlighted-text
-                                 :size     :paragraph-2
-                                 :weight   :medium}
-                                (i18n/label :t/terms-of-service)]
-                               [quo/text
-                                {:style style/plain-text
-                                 :size  :paragraph-2}
-                                " " (i18n/label :t/and) " "]
-                               [quo/text
-                                {:on-press #(rf/dispatch [:show-bottom-sheet
-                                                          {:content privacy/privacy-statement
-                                                           :shell?  true}])
-                                 :style    style/highlighted-text
-                                 :size     :paragraph-2
-                                 :weight   :medium}
-                                (i18n/label :t/intro-privacy-policy)]]]
-       :button-one-label     (i18n/label :t/sync-or-recover-profile)
-       :button-one-props     {:type                :dark-grey
-                              :disabled?           (not terms-accepted?)
-                              :accessibility-label :already-use-status-button
-                              :on-press            (fn []
-                                                     (when-let [blur-show-fn @overlay/blur-show-fn-atom]
-                                                       (blur-show-fn))
-                                                     (rf/dispatch
-                                                      [:open-modal
-                                                       :screen/onboarding.sync-or-recover-profile]))}
-       :button-two-label     (i18n/label :t/create-profile)
-       :button-two-props     {:accessibility-label :new-to-status-button
-                              :disabled? (not terms-accepted?)
-                              :on-press
-                              (fn []
-                                (when-let [blur-show-fn @overlay/blur-show-fn-atom]
-                                  (blur-show-fn))
-                                (rf/dispatch
-                                 [:open-modal :screen/onboarding.new-to-status]))}}]
+      (cond->
+        {:container-style      (style/bottom-actions-container (safe-area/get-bottom))
+         :actions              :two-vertical-actions
+         :description          :top
+         :description-top-text [terms terms-accepted? set-terms-accepted?]}
+        from-v1-without-terms-accepted?
+        (assoc
+         :actions          :one-action
+         :button-one-label (i18n/label :t/explore-the-new-status)
+         :button-one-props {:disabled?           (not terms-accepted?)
+                            :accessibility-label :explore-new-status
+                            :on-press            explore-new-status})
+
+        (not from-v1-without-terms-accepted?)
+        (assoc
+         :actions          :two-vertical-actions
+         :button-one-label (i18n/label :t/sync-or-recover-profile)
+         :button-one-props {:type                :dark-grey
+                            :disabled?           (not terms-accepted?)
+                            :accessibility-label :already-use-status-button
+                            :on-press            sync-or-recover-profile}
+         :button-two-label (i18n/label :t/create-profile)
+         :button-two-props {:accessibility-label :new-to-status-button
+                            :disabled?           (not terms-accepted?)
+                            :on-press            create-profile}))]
      [overlay/view]]))

--- a/src/status_im/contexts/profile/data_store.cljs
+++ b/src/status_im/contexts/profile/data_store.cljs
@@ -1,0 +1,5 @@
+(ns status-im.contexts.profile.data-store)
+
+(defn accepted-terms?
+  [accounts]
+  (some #(:hasAcceptedTerms %) accounts))

--- a/src/status_im/contexts/profile/effects.cljs
+++ b/src/status_im/contexts/profile/effects.cljs
@@ -1,0 +1,14 @@
+(ns status-im.contexts.profile.effects
+  (:require
+    [native-module.core :as native-module]
+    [promesa.core :as promesa]
+    [taoensso.timbre :as log]
+    [utils.re-frame :as rf]))
+
+(rf/reg-fx :effects.profile/accept-terms
+ (fn [{:keys [on-success]}]
+   (-> (native-module/accept-terms)
+       (promesa/then (fn []
+                       (rf/call-continuation on-success)))
+       (promesa/catch (fn [error]
+                        (log/error "Failed to accept terms" {:error error}))))))

--- a/src/status_im/contexts/profile/events.cljs
+++ b/src/status_im/contexts/profile/events.cljs
@@ -4,10 +4,12 @@
     [legacy.status-im.multiaccounts.update.core :as multiaccounts.update]
     [native-module.core :as native-module]
     [status-im.config :as config]
+    [status-im.contexts.profile.data-store :as profile.data-store]
     [status-im.contexts.profile.edit.accent-colour.events]
     [status-im.contexts.profile.edit.bio.events]
     [status-im.contexts.profile.edit.header.events]
     [status-im.contexts.profile.edit.name.events]
+    status-im.contexts.profile.effects
     status-im.contexts.profile.login.events
     [status-im.contexts.profile.rpc :as profile.rpc]
     [utils.re-frame :as rf]))
@@ -42,26 +44,27 @@
 (rf/reg-event-fx
  :profile/get-profiles-overview-success
  (fn [{:keys [db]} [{:keys [accounts] {:keys [userConfirmed enabled]} :centralizedMetricsInfo}]]
-   (let [db-with-settings (assoc db
-                                 :centralized-metrics/user-confirmed? userConfirmed
-                                 :centralized-metrics/enabled?        enabled)]
-     (if (seq accounts)
-       (let [profiles          (reduce-profiles accounts)
-             {:keys [key-uid]} (first (sort-by :timestamp > (vals profiles)))]
-         {:db (if key-uid
-                (-> db-with-settings
-                    (assoc :profile/profiles-overview profiles)
-                    (update :profile/login #(select-profile % key-uid)))
-                db-with-settings)
-          :fx [[:dispatch [:update-theme-and-init-root :screen/profile.profiles]]
-               (when (and key-uid userConfirmed)
-                 [:effects.biometric/check-if-available
-                  {:key-uid    key-uid
-                   :on-success (fn [auth-method]
-                                 (rf/dispatch [:profile.login/check-biometric-success key-uid
-                                               auth-method]))}])]})
-       {:db db-with-settings
-        :fx [[:dispatch [:update-theme-and-init-root :screen/onboarding.intro]]]}))))
+   (let [db-with-settings  (assoc db
+                                  :centralized-metrics/user-confirmed? userConfirmed
+                                  :centralized-metrics/enabled?        enabled)
+         profiles          (reduce-profiles accounts)
+         {:keys [key-uid]} (first (sort-by :timestamp > (vals profiles)))
+         new-db            (cond-> db-with-settings
+                             (seq profiles)
+                             (assoc :profile/profiles-overview profiles)
+
+                             key-uid
+                             (update :profile/login #(select-profile % key-uid)))]
+     {:db new-db
+      :fx (if (profile.data-store/accepted-terms? accounts)
+            [[:dispatch [:update-theme-and-init-root :screen/profile.profiles]]
+             (when (and key-uid userConfirmed)
+               [:effects.biometric/check-if-available
+                {:key-uid    key-uid
+                 :on-success (fn [auth-method]
+                               (rf/dispatch [:profile.login/check-biometric-success key-uid
+                                             auth-method]))}])]
+            [[:dispatch [:update-theme-and-init-root :screen/onboarding.intro]]])})))
 
 (rf/reg-event-fx
  :profile/update-setting-from-backup
@@ -82,3 +85,8 @@
     :messages-from-contacts-only
     (not (get-in db [:profile/profile :messages-from-contacts-only]))
     {})))
+
+(rf/reg-event-fx :profile/explore-new-status
+ (fn []
+   {:fx [[:effects.profile/accept-terms
+          {:on-success [:navigate-to :screen/profile.profiles]}]]}))

--- a/src/status_im/subs/profile.cljs
+++ b/src/status_im/subs/profile.cljs
@@ -9,6 +9,7 @@
     [re-frame.core :as re-frame]
     [status-im.common.pixel-ratio :as pixel-ratio]
     [status-im.constants :as constants]
+    [status-im.contexts.profile.data-store :as profile.data-store]
     [status-im.contexts.profile.utils :as profile.utils]
     [utils.security.core :as security]))
 
@@ -17,6 +18,18 @@
  :<- [:profile/profile]
  (fn [{:keys [customization-color]}]
    (or customization-color constants/profile-default-color)))
+
+(re-frame/reg-sub :profile/accepted-terms?
+ :<- [:profile/profile]
+ (fn [{:keys [hasAcceptedTerms]}]
+   hasAcceptedTerms))
+
+;; A profile can only be created without accepting terms in Status v1.
+(re-frame/reg-sub :profile/from-status-v1-without-terms-accepted?
+ :<- [:profile/profiles-overview]
+ (fn [profiles-overview]
+   (and (seq profiles-overview)
+        (not (profile.data-store/accepted-terms? (vals profiles-overview))))))
 
 (re-frame/reg-sub
  :profile/currency

--- a/src/tests/test_utils.cljs
+++ b/src/tests/test_utils.cljs
@@ -70,6 +70,9 @@
    {:initializeApplication
     (fn [request callback]
       (callback (.initializeApplication native-status request)))
+    :acceptTerms
+    (fn [callback]
+      (callback (.acceptTerms native-status)))
     :createAccountAndLogin
     (fn [request] (.createAccountAndLogin native-status request))
     :restoreAccountAndLogin

--- a/status-go-version.json
+++ b/status-go-version.json
@@ -4,6 +4,6 @@
     "owner": "status-im",
     "repo": "status-go",
     "version": "release/0.182.x",
-    "commit-sha1": "4a18c85c3c1d58ea6c8493c46bfd2ed5772b1386",
-    "src-sha256": "04fgykwk44r7f16bfxlkpl9kgnl7yssfyycqnddwszinnnlnfmpl"
+    "commit-sha1": "14c996158cf1d651d44808d51686fdbfb2eb3b39",
+    "src-sha256": "02dnz3327kz8cnhqp6cgcmvqvhcdc941iic7jz2v35hck221vbkg"
 }

--- a/translations/en.json
+++ b/translations/en.json
@@ -986,6 +986,7 @@
   "expand-all": "Expand all",
   "experienced-web3": "Experienced in Web3?",
   "explore-the-decentralized-web": "Explore and interact with the decentralized web",
+  "explore-the-new-status": "Explore the new Status",
   "export-account": "Export account",
   "export-key": "Export private key",
   "external-link": "External link",

